### PR TITLE
Multiple organizers for an event

### PR DESCRIPTION
**Pull request by jbrinley**
Add UI to the event admin to add multiple organizers to the event

http://central.tri.be/issues/33808

### CODE REVIEW

File: src/Tribe/Admin/Organizer_Chooser_Meta_Box.php [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+	 * the middle of creating a new event.
+	 *
+	 * @param null $event
+	 */
+	protected function get_event( $event = null ) {
+		global $post;
+
+		if ( $event === null ) {
+			$this->event = $post;
+		} elseif ( is_a( $event, 'WP_Post' ) ) {
```
Comment (borkweb) [inline src/Tribe/Admin/Organizer_Chooser_Meta_Box.php]: `is_a()` was briefly deprecated in PHP 5.2, thus throws deprecation notices for that version. Switching this to `$event instanceof WP_Post` should resolve that.

File: src/Tribe/Admin/Organizer_Chooser_Meta_Box.php [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+			$this->event = $event;
+		} else {
+			$this->event = new WP_Post( (object) array( 'ID' => 0 ) );
+		}
+	}
+
+	public function render() {
+		$this->render_dropdowns();
+		$this->render_add_organizer_button();
+		include( $this->tribe->pluginPath . 'admin-views/new-organizer-meta-section.php' );
```
Comment (borkweb) [inline src/Tribe/Admin/Organizer_Chooser_Meta_Box.php]: Superfluous parens - change it if you want. Not a requirement, obviously.

File: src/Tribe/Admin/Organizer_Chooser_Meta_Box.php [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+		if ( $event === null ) {
+			$this->event = $post;
+		} elseif ( is_a( $event, 'WP_Post' ) ) {
+			$this->event = $event;
+		} else {
+			$this->event = new WP_Post( (object) array( 'ID' => 0 ) );
+		}
+	}
+
+	public function render() {
```
Comment (borkweb) [inline src/Tribe/Admin/Organizer_Chooser_Meta_Box.php]: Add a docblock comment for this method?

File: src/Tribe/Admin/Organizer_Chooser_Meta_Box.php [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+		?><script type="text/template" id="tmpl-tribe-select-organizer"><?php $this->single_organizer_dropdown(0); ?></script><?php
+
+		foreach ( $current_organizers as $organizer_id ) {
+			echo '<tbody>';
+			$this->single_organizer_dropdown( $organizer_id );
+			echo '</tbody>';
+		}
+	}
+
+	protected function single_organizer_dropdown( $organizer_id ) {
```
Comment (borkweb) [inline src/Tribe/Admin/Organizer_Chooser_Meta_Box.php]: Add a docblock comment for this method?

Comment (borkweb) [inline src/Tribe/Admin/Organizer_Chooser_Meta_Box.php]: Rinse and repeat for other methods missing docblocks comments.

File: src/Tribe/Admin/Organizer_Chooser_Meta_Box.php [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+		</tr>
+	<?php
+
+	}
+
+	protected function edit_organizer_link( $organizer_id ) {
+		?>
+		<div class="edit-organizer-link"><a
+				<?php if ( empty( $organizer_id ) ) { ?> style="display:none;"<?php } ?>
+				data-admin-url="<?php echo admin_url( 'post.php?action=edit&post=' ); ?>"
```
Comment (borkweb) [inline src/Tribe/Admin/Organizer_Chooser_Meta_Box.php]: `esc_url` the `admin_url`s on this line and the next.

File: src/views/modules/meta/organizer.php [hunk 6]
```diff
@@ -8,32 +8,41 @@
...
-		<dd class="fn org"> <?php echo tribe_get_organizer() ?> </dd>
+		<?php foreach ( $organizer_ids as $organizer ) { ?>
+			<dd class="fn org"> <?php echo tribe_get_organizer( $organizer ) ?> </dd>
+		<?php } ?>
+
+		<?php if ( !$multiple ) { // only show organizer details if there is one ?>
+
+			<?php if ( ! empty( $phone ) ): ?>
+				<dt> <?php esc_html_e( 'Phone:', 'tribe-events-calendar' ) ?> </dt>
+				<dd class="tel"> <?php echo $phone ?> </dd>
```
Comment (borkweb) [inline src/views/modules/meta/organizer.php]: Escape `$phone`

File: src/views/modules/meta/organizer.php [hunk 7]
```diff
@@ -8,32 +8,41 @@
...
+			<?php endif ?>
 
-		<?php if ( ! empty( $phone ) ): ?>
-			<dt> <?php esc_html_e( 'Phone:', 'tribe-events-calendar' ) ?> </dt>
-			<dd class="tel"> <?php echo $phone ?> </dd>
-		<?php endif ?>
+			<?php if ( ! empty( $email ) ): ?>
+				<dt> <?php esc_html_e( 'Email:', 'tribe-events-calendar' ) ?> </dt>
+				<dd class="email"> <?php echo $email ?> </dd>
```
Comment (borkweb) [inline src/views/modules/meta/organizer.php]: Escape `$email`

File: src/views/modules/meta/organizer.php [hunk 8]
```diff
@@ -8,32 +8,41 @@
...
+			<?php endif ?>
 
-		<?php if ( ! empty( $email ) ): ?>
-			<dt> <?php esc_html_e( 'Email:', 'tribe-events-calendar' ) ?> </dt>
-			<dd class="email"> <?php echo $email ?> </dd>
-		<?php endif ?>
+			<?php if ( ! empty( $website ) ): ?>
+				<dt> <?php esc_html_e( 'Website:', 'tribe-events-calendar' ) ?> </dt>
+				<dd class="url"> <?php echo $website ?> </dd>
```
Comment (borkweb) [inline src/views/modules/meta/organizer.php]: Escape `$website`

**Comment by borkweb** [comment]
Woo! This is pretty awesome!

File: src/Tribe/Admin/Organizer_Chooser_Meta_Box.php (hunk 2 above)
Comment (jbrinley) [inline src/Tribe/Admin/Organizer_Chooser_Meta_Box.php]: I'm rather happy with my parentheses, thank you. :)

**Comment by jbrinley** [comment]
I've addressed all of the above comments.

**Comment by borkweb** [comment]
Woo! This looks great! Let's get this merged.

:gem: :gem:

**Comment by borkweb** [comment]
@zbtirrell: because of the funky TBD wording in the [Central ticket](http://central.tri.be/issues/33808), I'll leave it up to you if we want to merge it right now. I can't think of a reason not to.

**Comment by zbtirrell** [comment]
I think this is wonderful.    Considering this has been in use for Stanford, I'm going to skip the pre-merge QA and we'll get it in smoketest.  Merging and moving the issue to current release.

**Pull request merged**